### PR TITLE
Examples

### PR DESCRIPTION
**Pull request by Robinlovelace**

### CODE REVIEW

**Comment by Robinlovelace** [comment]
Heads-up @dabreegster I think this is ready to go. I took a read about examples and the simple Rust example in the examples folder seems to work (there's also an R example I used for checking, not sure if that's the best place for it).

Initially parts of #34 were in there too but that's a bigger job for so have split out into a separate PR.

File: examples/clockboard-example.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// See https://github.com/zonebuilders/zonebuilder-rust/issues/23
+extern crate zonebuilder;
```
Comment (dabreegster) [inline examples/clockboard-example.rs]: I think you can just remove this line; `extern crate` is something that an older version of Rust needed. I think you only use it now to import macros from another crate.

File: examples/r-tests.R [hunk 2]
```diff
@@ -0,0 +1,8 @@
+setwd("~/github-orgs/zonebuilders/zonebuilder-rust/")
```
Comment (dabreegster) [inline examples/r-tests.R]: It seems super weird to assume this is where someone's cloned the repo. Can the R script detect its own location and know that the crate root is one dir up?

File: src/lib.rs [hunk 3]
```diff
@@ -161,12 +151,12 @@ fn clockpoly(
...
-    let ti = 1 + (seg + 1) * nv;
-    let seq = fi..ti;
+    let from_iterator = seg * nv;
+    let to_iterator = 1 + (seg + 1) * nv;
+    let seq = from_iterator..to_iterator;
     // Angle offset so first segment is North
     let o = std::f64::consts::PI / (num_segments as f64);
-    let seq_reverse = (fi..ti).rev();
+    let seq_reverse = (from_iterator..to_iterator).rev();
     for i in seq {
```
Comment (dabreegster) [inline src/lib.rs]: There's lots of intermediate variables defined here. Personally, I would find it a bit easier to read if some of the ones used only once are just inlined in that single place. So here, why not `for i in from_iterator..to_iterator`?

**Review by dabreegster — APPROVED** [review]

File: src/lib.rs (hunk 3 above)
Comment (Robinlovelace) [reply src/lib.rs]: Agreed. This is now more concise, good style finding a balance between readability and conciseness, or moving in that direction:

https://github.com/zonebuilders/zonebuilder-rust/pull/29/commits/734dff9b743b5115b17b5b1ee2a50defab68e869

File: examples/clockboard-example.rs (hunk 1 above)
Comment (Robinlovelace) [reply examples/clockboard-example.rs]: :+1:

File: examples/r-tests.R (hunk 2 above)
Comment (Robinlovelace) [reply examples/r-tests.R]: :+1:

Comment (Robinlovelace) [reply examples/r-tests.R]: Detail on this one: I added it once for the 'reprex' package to run and show the output. Good news, this is now redundant also as the developers of the package have added a `wd` argument: https://reprex.tidyverse.org/reference/reprex.html

It was left in by mistake in any case, well spotted, definitely we're better off without this!

**Pull request merged**